synapse/handlers/e2e_keys.py

```diff
@@ -1107,10 +1107,6 @@ async def _process_self_signatures(
                     # can skip it, since it should be exactly the same
                     continue
 
-                _check_device_signature(
-                    user_id, self_signing_verify_key, device, stored_device
-                )
```

Comment:
There's one more caller of this in `_check_master_key_signature` -- is that use-case valid?

Reply:
As I mentioned in a comment, I left the validation in /keys/upload and I do belive that call is necessary.
(as in cross signing master key is signed by the device keys)
master key -> self signing key -> device key -> master key

Reply:
Ah-ha, I didn't connect that `/keys/upload` called into that directly. 👍